### ISSUE TITLE
Lexer: Chaining: Leading dot not working with `do`

### ISSUE THREAD

**Issue by darky** [jashkenas/coffeescript#3736]
``` coffeescript
do ->
  "foo qwe"
.replace "qwe", "bar"
```

**Comment by GeoffreyBooth** [comment]
This was fixed, in the sense that it compiles now, via #4665; but the new output isn’t what you want:

```coffee
do ->
  "foo qwe"
.replace "qwe", "bar"
```
___
```js
(function() {
  return "foo qwe";
}).replace("qwe", "bar")();
```
Note that the `()`, thanks to the `do`, is at the end, rather than before `.replace`. Apparently this has always been the way that `do` works with chaining:
```coffee
do foo
.bar
# foo.bar()
```
I’m not sure if anyone uses `do` with chained functions like this, but if we were to change it so that `do` only applied to the immediate next function, that would be a breaking change. But the current behavior makes sense, in that the chained version is supposed to be treated the same way as the unchained version:
```coffee
do foo.bar
# foo.bar()
```
So having `do foo\n.bar` become `foo().bar` would be inconsistent. I’m afraid you’ll just need to use parentheses to produce what you’re intending:
```coffee
(do ->
  "foo qwe"
).replace "qwe", "bar"
```

**Comment by helixbass** [comment]
@GeoffreyBooth I think the current compilation (as of #4665) is quite unintuitive/unexpected. And this chained `do <IIFE>` construct is one I'd like to be able to use. So I opened #4666, see there for more explanation but basically I think the inconsistency it would introduce between chained `do <IIFE>` vs chained `do <non-IIFE>` is overshadowed by the (now-dangerous since it no longer fails to compile) unexpected compilation of the useful chained `do <IIFE>` construct

**Comment by GeoffreyBooth** [comment]
Yeah I was reconsidering this today, I think you're right.

**Comment by GeoffreyBooth** [comment]
Fixed further via #4672.
